sdk/storage/storage-blob/src/highlevel.node.ts

```diff
@@ -377,3 +443,38 @@ export async function uploadStreamToBlockBlob(
 
   return blockBlobURL.commitBlockList(blockList, options);
 }
+
+type UploadStreamToBlockBlobUrlOptions = IUploadStreamToBlockBlobOptions & CredentialOptions & INewPipelineOptions
+
+/**
+ * ONLY AVAILABLE IN NODE.JS RUNTIME.
+ *
+ * Uploads a Node.js Readable stream into block blob given a url to that block blob.
+ * This method assumes that the container already exists.
+ *
+ * PERFORMANCE IMPROVEMENT TIPS:
+ * * Input stream highWaterMark is better to set a same value with bufferSize
+ *    parameter, which will avoid Buffer.concat() operations.
+ *
+ * @export
+ * @param {Readable} stream Node.js Readable stream
+ * @param {BlockBlobURL} blockBlobURL A BlockBlobURL instance
+ * @param {number} bufferSize Size of every buffer allocated, also the block size in the uploaded block blob
+ * @param {number} maxBuffers Max buffers will allocate during uploading, positive correlation
+ *                            with max uploading concurrency
+ * @param {UploadStreamToBlockBlobUrlOptions} [options] IUploadStreamToBlockBlobOptions & CredentialOptions & INewPipelineOptions.
+ *                                            If credential options is not specified {@link AnonymousCredential} is used.
+ * @returns {Promise<BlobUploadCommonResponse>}
+ */
+export async function uploadStreamToBlockBlobUrl(
+  stream: Readable,
+  url: string,
+  bufferSize: number,
```

Comment:
Can `bufferSize` and `maxBuffers` have defaults set? These seem like nuanced parameters that developers may not know how to provide a high-level API.

Reply:
They are used for parallel uploading. Looks like size for each segment and max number of segments in flight. I picked 4MB and 20 based on the tests.